### PR TITLE
Detect Win

### PR DESCRIPTION
**Pull request by mkrump**
https://trello.com/c/0KqHBk6G/31-2-ability-to-detect-a-win-even-when-board-is-full

### CODE REVIEW

File: Tests/GameTests.cs [hunk 1]
```diff
@@ -19,25 +18,31 @@ public GameTests()
...
-            mockBoard.SetupProperty(board => board.Contents, emptyBoard);
-
             playerList = new List<IPlayer> {mockPlayer1.Object, modkPlayer2.Object};
-            game = new Game(mockBoard.Object, playerList);
         }
 
         [Fact]
         public void CurrentPlayer()
         {
+            var emptyBoard = Enumerable.Repeat("", 9).ToList();
```
Comment (andarcabrera) [inline Tests/GameTests.cs]: When I look at the setup for these tests, it seems that there is duplication. Can you introduce a test helper and make it more DRY? Maybe you can use a beforeEach or beforeAll

Comment (andarcabrera) [inline Tests/GameTests.cs]: I'm referring to this block of code:

```var emptyBoard = Enumerable.Repeat("", 9).ToList();
 +            var mockBoard = new Mock<IBoard>();
 +            mockBoard.Setup(board => board.GridSize).Returns(3);
 +            mockBoard.SetupProperty(board => board.Contents, emptyBoard);
 +            var game = new Game(mockBoard.Object, playerList);```

File: TicTacToe/Games/Game.cs [hunk 2]
```diff
@@ -70,9 +84,49 @@ public List<int> OpenSquares()
...
+            IsGameOver = Winner != null || IsTie;
+        }
+
+        public void CheckIfTie()
+        {
+            if (IsTie || Winner != null) return;
+            IsTie = OpenSquares().Count == 0;
+        }
+
+        private List<HashSet<int>> GenerateWinningCombos()
```
Comment (andarcabrera) [inline TicTacToe/Games/Game.cs]: The Game class is getting pretty big. Does it have to create the winning combos itself or can we introduce that  as a dependency an keep it more open-ended? What if I want to initiate the game with different rules, that say that 1-2-5 is a winning combination?

Comment (andarcabrera) [inline TicTacToe/Games/Game.cs]: this `GenerateWinningCombos` method can be tidied up a bit.  Maybe one method per winning combo would be easier to follow

File: TicTacToe/UIConfigFiles/fancy.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
     "MenuName": "fancy_ui",
-    "TieMessage": "-- Tie ---",
+    "TieMessage": "$$$ Tie $$$",
+    "WinMessage": "$$$ Win $$$",
```
Comment (andarcabrera) [inline TicTacToe/UIConfigFiles/fancy.json]: 👍 for personalizing the winning message depending on UI choice

**Comment by mkrump** [comment]
#12 includes these changes + some additional suggestions regarding structure of project.

**Pull request closed without merging**